lib/mongoid/history/trackable.rb

```diff
@@ -74,21 +74,41 @@ def history_tracks
         #  undo :from => 1, :to => 5
         #  undo 4
         #  undo :last => 10
-        def undo!(modifier = nil, options_or_version = nil)
+        def undo(modifier = nil, options_or_version = nil)
           versions = get_versions_criteria(options_or_version).to_a
           versions.sort! { |v1, v2| v2.version <=> v1.version }
 
           versions.each do |v|
-            undo_attr = v.undo_attr(modifier)
-            if Mongoid::History.mongoid3? # update_attributes! not bypassing rails 3 protected attributes
-              assign_attributes(undo_attr, without_protection: true)
-              save!
-            else # assign_attributes with 'without_protection' option does not work with rails 4/mongoid 4
-              update_attributes!(undo_attr)
-            end
+            # NOTE: not sure how this works with different versions on Mongoid,
+            #       works for rails 4/ mongoid 4
+            undo_attr       = v.undo_attr(modifier)
+            self.attributes = undo_attr
```

Comment:
I think you don't need to assign `undo_attr`, just do `self.attributes = v....`.